content/docs/userspace/gall-guide/types.md

```diff
@@ -0,0 +1,221 @@
+of effects and a new state.
+
+## `path`
+
+The `path` type is formally defined as:
+
+```hoon
++$  path  (list knot)
+```
+
+A knot is a `@ta` text atom (see the [Strings guide](/docs/hoon/guides/strings)
+for details), so a `path` is just a list of text. Rather than having to write
+`[~.foo ~.bar ~.baz ~]` though, it has its own syntax which looks like
+`/foo/bar/baz`.
+
+A `path` is just a general type used for several different purposes throughout
+Arvo. Its elements have no inherent significance, it depends on the context. In
+a Gall agent, a `path` is most commonly a subscription path - you might
+subscribe for updates to `/foo/bar` on another agent, or another agent might
+subscribe to `/baz` on your agent.
+
+A `path` might just be a series of fixed elements like `/foo/bar`, but some
+elements might also be variable and encode atom auras, or some other format. For
+example, you might like to include a date in the path like
+`/updates/~2021.10.31..07.24.27..db68`. Other agents might create the path by
+doing something like:
+
+```hoon
+/update/(scot %da now.bowl)
+```
+
+Then, when you get a subscription request, you might do something like:
+
+```hoon
+?+    path  (on-watch:def path)
+    [%updates @ ~]
+  =/  date=@da  (slav %da i.t.path)
+  ...(rest of code)...
+```
+
+See the [Encoding in text](/docs/hoon/guides/strings#encoding-in-text) and
+[Decoding from text](/docs/hoon/guides/strings#decoding-from-text) sections of
+the Strings guide for more information on dealing with atoms encoded in strings.
+
+Aside from using function calls when constructing a `path` as demonstrated
+above, you can also insert text you're previously stored with `=/` or what have
+you, simply by enclosing them in brackets. For example, in the dojo:
+
+```
+> =const 'bar'
+> `path`/foo/[const]/baz
+/foo/bar/baz
+```
+
+## `wire`
+
+The type of a wire is formally defined as:
+
+```hoon
++$  wire  path
+```
+
+So, a `wire` is just a [`path`](#path), type-wise they're exactly the same. The
+reason there's a separate `wire` type is just to differentiate their purpose. A
+`wire` is a path for responses to requests an agent initiates. If you subscribe
+to the `path` `/some/path` on another agent, you also specify `/some/wire`.
+Then, when that agent sends out updates to subscribers of `/some/path`, your
+agent receives them on `/some/wire`.
+
+The `wire` is specified in the second argument of a `%pass` `card`. It's used
+for anything you can `%pass`, such as `%poke`s, `%watch`es, and `%arvo` notes.
+For example:
+
+```hoon
+[%pass /this/is/wire %agent [~zod %foobar] %watch /this/is/path]
+::
+[%pass /this/is/wire %agent [~zod %foobar] %poke %foo !>('hello')]
+::
+[%pass /this/is/wire %arvo %b %wait (add now.bowl ~m1)]
+```
+
+The `on-agent` and `on-arvo` arms of the agent core include a `wire` in their
+respective sample. Responses from agents come in to the former, and responses
+from vanes come in to the latter.
```

Comment:
May also consider briefly mentioning the role of `wire`s in call stacks and `duct`s, linking to `/docs/arvo/overview#types`